### PR TITLE
Pull settings from config file, write/import via S3

### PR DESCRIPTION
**Pull request by bmtcril**
Apologies for the massive PR, this has been getting heavy iteration while I've been load testing for the last several weeks. The culmination of this work is a much faster, more reproducible, and more realistic set of data that exercises all of the existing Aspects Instructor Dashboard reports. 

The changes fall into a few buckets:

- Move settings to a config file format, there were so many it was becoming impossible to deal with on the command line
- Remove schema management, you must now pre-create the schema using Aspects
- Refactor settings to use explicit numbers of actors/enrollments/courses/blocks instead of randomly assigning them, and make them configurable for explicitly testing different factors of load
- Refactor actor usage, actors now have enrollment dates for each course, are shared across courses more effectively, and have fake PII / external ids.
- Update course block data, blocks now contain a fair approximation of section/subsection/unit as Aspects expects
- Update some xAPI statements to match current ERB output
- Add ability to write CSVs directly to S3 (1B events is ~213GB!)
- Add ability to import CSVs directly from S3 to ClickHouse for faster and more stable iteration on schema changes
- Add forum post statement type

### CODE REVIEW

File: xapi_db_load/backends/clickhouse_lake.py [hunk 1]
```diff
@@ -23,12 +22,18 @@ def __init__(
...
         self.port = db_port
         self.username = db_username
         self.database = db_name
+        self.event_sink_database = db_event_sink_name
         self.db_password = db_password
+        self.s3_key = s3_key
+        self.s3_secret = s3_secret
 
         self.event_raw_table_name = "xapi_events_all"
         self.event_table_name = "xapi_events_all_parsed"
```
Comment (Ian2012) [inline xapi_db_load/backends/clickhouse_lake.py:39]: As in Aspects both of these databases can be configurable, shouldn't those be configurable too?

File: xapi_db_load/backends/clickhouse_lake.py [hunk 2]
```diff
@@ -332,27 +176,127 @@ def insert_event_sink_block_data(self, courses):
...
+                    '{actor.bio}',
+                    '{actor.profile_image_uploaded_at}',
+                    '{actor.phone_number}',
+                    '{dump_id}',
+                    '{dump_time}'
+                )"""
+
+                out_profile.append(profile_row)
+            except Exception:
+                print(actor)
```
Comment (Ian2012) [inline xapi_db_load/backends/clickhouse_lake.py:234]: nit: remove print statement

Comment (Ian2012) [reply xapi_db_load/backends/clickhouse_lake.py:234]: I don't think we need this try - except block

File: xapi_db_load/backends/clickhouse_lake.py [hunk 3]
```diff
@@ -398,7 +342,7 @@ def do_queries(self, event_generator):
         )
 
         self._run_query_and_print(
-            "Count of enrollments for this learner",
+            "Count of enrollments for this actpr",
```
Comment (Ian2012) [inline xapi_db_load/backends/clickhouse_lake.py:345]: nit: actor

**Review by Ian2012 — COMMENTED** [review]
Overall looks good to me, still pending a testing with s3 and k8s

**Comment by Ian2012** [comment]
@bmtcril To avoid issues with updates on ERB, should we make obligatory to update this repo too whenever a contribution or change is done there?

**Comment by bmtcril** [comment]
@Ian2012 I don't think we need to block ERB on this project, it's generally ok for it to fall a little behind since it's not a validation tool, just a best approximation load test tool.

**Review by Ian2012 — APPROVED** [review]
Tested locally with s3, CSV, Clickhouse and Ralph backends

**Pull request merged**